### PR TITLE
Added datacenter filter to asset reports

### PR DESCRIPTION
**Pull request by kula1922**
In Assets -> Reports -> Status - model -> Only data center added datacenter filter

### CODE REVIEW

File: src/ralph_assets/views/report.py [hunk 1]
```diff
@@ -407,6 +413,24 @@ class ReportViewBase(AssetsBase):
...
+            'name': 'dc2',
+            'verbose_name': 'DC2',
+        },
+        {
+            'name': 'dc4',
+            'verbose_name': 'DC4',
+        },
+        {
+            'name': 'dc5',
+            'verbose_name': 'DC5',
```
Comment (quamilek) [inline src/ralph_assets/views/report.py]: Hard coded names is very ugly

File: src/ralph_assets/views/report.py [hunk 2]
```diff
@@ -231,13 +234,16 @@ def prepare(self, mode=None):
 
 
 class StatusModelReport(BaseReport):
+    with_datacenters = True
     slug = 'status-model'
     name = _('Status - model')
 
-    def prepare(self, mode=None):
+    def prepare(self, mode=None, dc=None):
```
Comment (quamilek) [inline src/ralph_assets/views/report.py]: BackOffice part of Assets not containing DC parameters.

File: src/ralph_assets/views/report.py [hunk 3]
```diff
@@ -457,6 +482,10 @@ def get_response(self, request, result):
         return self.report.get_response(request, result)
 
     def dispatch(self, request, *args, **kwargs):
+        try:
+            self.dc = DataCenter.objects.get(name=kwargs.pop('dc', None))
```
Comment (quamilek) [inline src/ralph_assets/views/report.py]: Why `kwargs.pop`?

File: src/ralph_assets/templates/assets/report_detail.html [hunk 4]
```diff
@@ -18,7 +18,18 @@
       <div class="pagination">
         <ul>
           {% for report_mode in modes %}
-            <li{% if mode == report_mode.name %} class="active"{% endif %}><a href="{% url report_detail report_mode.name slug %}">{{ report_mode.verbose_name }}</a></li>
+            <li{% if mode == report_mode.name %} class="active"{% endif %}><a href="{% if report_mode.name == 'dc' %}{% url report_detail report_mode.name dc slug %}{% else %}{% url report_detail report_mode.name slug %}{% endif %}">{{ report_mode.verbose_name }}</a></li>
```
Comment (quamilek) [inline src/ralph_assets/templates/assets/report_detail.html]: I suggest to display data centers filter as list to choosen by user. Select box must contains contains `all DCs` options.

File: src/ralph_assets/views/report.py (hunk 1 above)
Comment (vi4m) [inline src/ralph_assets/views/report.py]: yes, it should be generated from db

File: src/ralph_assets/views/report.py (hunk 2 above)
Comment (kula1922) [inline src/ralph_assets/views/report.py]: Yes and for them dc should be None

File: src/ralph_assets/templates/assets/report_detail.html (hunk 4 above)
Comment (kula1922) [inline src/ralph_assets/templates/assets/report_detail.html]: as we talked

File: src/ralph_assets/views/report.py [hunk 5]
```diff
@@ -408,6 +414,21 @@ class ReportViewBase(AssetsBase):
...
+    def datacenters(self):
+        datacenters = [
+            {
+                'name': 'all',
+                'verbose_name': 'All',
+            },
+        ]
+        return datacenters + [
+            {
+                'name': dc.name.lower(),
```
Comment (mkurek) [inline src/ralph_assets/views/report.py]: why name, not id (ex. what if name contains some url-messing characters, ex. whitespace, / etc)?

**Comment by mkurek** [comment]
:ok: 
(@quamilek please review it again)

File: src/ralph_assets/tests/functional/tests_reports.py [hunk 6]
```diff
@@ -140,7 +140,7 @@ def test_report_linked_empty_report(self):
             asset.device_info.save()
 
         report = LinkedDevicesReport()
-        result = report.execute(None)
+        result = report.execute(None, None)
```
Comment (quamilek) [inline src/ralph_assets/tests/functional/tests_reports.py]: Default param ``dc=None`` in function ``execute`` looks better :smile:

File: src/ralph_assets/templates/assets/report_detail.html (hunk 4 above)
Comment (quamilek) [inline src/ralph_assets/templates/assets/report_detail.html]: This line is too long.

**Comment by ar4s** [comment]
:ok:

**Pull request merged**